app/src/main/java/com/vero/tasky/agenda/domain/worker/SyncAgendaWorker.kt

```diff
@@ -0,0 +1,44 @@
+package com.vero.tasky.agenda.domain.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.vero.tasky.agenda.domain.repository.AgendaRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import retrofit2.HttpException
+
+@HiltWorker
+class SyncAgendaWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val agendaRepository: AgendaRepository
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+        val deletedEventIds = inputData.getStringArray(DELETED_EVENT_IDS) ?: emptyArray()
+        val deletedTaskIds = inputData.getStringArray(DELETED_TASK_IDS) ?: emptyArray()
+        val deletedReminderIds = inputData.getStringArray(DELETED_REMINDER_IDS) ?: emptyArray()
```

Comment:
If you are using an array vs list, because of this

You can do 2 things

1- Just convert from list to array before passing the ids to here
2- You are injecting the repository. You can simply get all the deleted info using it (or you can inject the dao and do it using that)